nixos/modules/services/databases/mysql.nix

```diff
@@ -406,7 +420,7 @@ in
 
                 ${concatMapStrings (user:
                   ''
-                    ( echo "CREATE USER IF NOT EXISTS '${user.name}'@'localhost' IDENTIFIED WITH ${if isMariaDB then "unix_socket" else "auth_socket"};"
+                    ( echo "CREATE USER IF NOT EXISTS '${user.name}'@'localhost' ${optionalString  (user.authentication_option == "password") ''IDENTIFIED BY PASSWORD '${user.password_hash}' ''} ${optionalString  (user.authentication_option == "socket") ''IDENTIFIED WITH  ${if isMariaDB then "unix_socket" else "auth_socket"} ''};"
```

Comment:
A few issues:
- line too long, should fit 100 columns
- what happens when password is changed? As I understand it, user already exists so his password will silently remain the same. This is bad. Please do run ALTER USER to change password hash when it was changed in NixOS config
- same for `authentication_option`